src/libpsl-native/src/setdate.h

```diff
@@ -17,6 +17,6 @@ typedef struct setDateInfo
     int32_t DST;
 } SetDateInfo;
 
-int32_t SetDate(const SetDateInfo &info);
+int32_t SetDate(const SetDateInfo info);
 
```

Comment:
No, the reference was not being marshaled as a pointer, it was being marshaled as a value (and then likely only worked because the compiler ignored the pass-by-reference in the extern C declaration).

```
[DllImport(psLib, CharSet = CharSet.Ansi, SetLastError = true)]
internal static extern int SetDate(SetDateInfoInternal info);
```

And looking at this further the `SetDateInfoInternal` is a class in C# with a pair of functions in it... I think this is working by luck, it seems broken to me. @palladia do you know why it works as-is currently?